src/services/gcs/backend.rs

```diff
@@ -0,0 +1,706 @@
+                e
+            })?;
+            let j_object: Value =
+                de::from_reader(v.reader()).map_err(|e| other(ObjectError::new("stat", &p, e)))?;
+
+            if let Some(Value::Number(content_length)) = j_object.get("size") {
+                if content_length.is_u64() {
+                    m.set_content_length(content_length.as_u64().unwrap());
+                }
+            }
+
+            if let Some(Value::String(etag)) = j_object.get("etag") {
+                m.set_etag(etag.as_str());
+            }
+
+            if let Some(Value::String(last_modified)) = j_object.get("last_modified") {
+                let datetime =
+                    OffsetDateTime::parse(last_modified.as_str(), &Rfc3339).map_err(|e| {
+                        error!("GCS backend failed to parse datetime in stat");
+                        other(ObjectError::new("stat", &p, e))
+                    })?;
+                m.set_last_modified(datetime);
+            }
+
+            if let Some(Value::String(md5_hash)) = j_object.get("md5Hash") {
+                m.set_content_md5(md5_hash.as_str());
+            }
+
+            if p.ends_with('/') {
+                m.set_mode(ObjectMode::DIR);
+            } else {
+                m.set_mode(ObjectMode::FILE);
+            };
+
+            debug!("object {} stat finished: {:?}", &p, m);
+            Ok(m)
+        } else if resp.status() == StatusCode::NOT_FOUND && p.ends_with('/') {
+            let mut m = ObjectMetadata::default();
+            m.set_mode(ObjectMode::DIR);
+
+            debug!("object {} stat finished", &p);
+            Ok(m)
+        } else {
+            let er = parse_error_response(resp).await?;
+            let e = parse_error("stat", args.path(), er);
+            Err(e)
+        }
+    }
+
+    #[trace("delete")]
+    async fn delete(&self, args: &OpDelete) -> Result<()> {
+        increment_counter!("opendal_gcs_delete_requests");
+
+        let p = self.get_abs_path(args.path());
+        debug!("object {} delete start", &p);
+
+        let resp = self.delete_object(&p).await?;
+
+        if resp.status().is_success() {
+            debug!("object {} delete finished", &p);
+            Ok(())
+        } else {
+            let er = parse_error_response(resp).await?;
+            let err = parse_error("delete", args.path(), er);
+            Err(err)
+        }
+    }
+
+    #[trace("list")]
+    async fn list(&self, args: &OpList) -> Result<DirStreamer> {
+        increment_counter!("opendal_gcs_list_requests");
+
+        let mut path = self.get_abs_path(args.path());
+        // Make sure list path is endswith '/'
+        if !path.ends_with('/') && !path.is_empty() {
+            path.push('/')
+        }
+        debug!("object {} list start", &path);
+
+        Ok(Box::new(DirStream::new(Arc::new(self.clone()), &path)))
+    }
+
+    fn presign(&self, _args: &OpPresign) -> Result<PresignedRequest> {
+        unimplemented!()
```

Comment:
Please remove this function directly (so we will fall back to default implementations).

As described in `Accessor`:

https://github.com/datafuselabs/opendal/blob/69c257b6a83f6cfb8c833f810a6b9d958738dcbd/src/accessor.rs#L136-L147